src/components/organisms/DashboardPane/DashboardPane.tsx

```diff
@@ -99,6 +113,46 @@ const DashboardPane = () => {
     }
   }, [activeMenuItemRef]);
 
+  const compareNamespaces = useCallback(
+    (namespace?: string) => {
+      if (clusterConnectionOptions.current.lastNamespaceLoaded === '<all>') {
+        return true;
+      }
+      if (clusterConnectionOptions.current.lastNamespaceLoaded === '<not-namespaced>') {
+        return !namespace;
+      }
+      return clusterConnectionOptions.current.lastNamespaceLoaded === namespace;
+    },
+    [clusterConnectionOptions]
+  );
+
+  const getResources = useCallback(
+    (kind: string) => {
+      return Object.values(clusterResourceMeta).filter(r => r.kind === kind && compareNamespaces(r.namespace));
+    },
+    [clusterResourceMeta, compareNamespaces]
+  );
+
+  const getResourceCount = useCallback((kind: string) => getResources(kind).length, [getResources]);
+
+  const getProblemCount = useCallback(
+    (kind: string, level: 'error' | 'warning') => {
+      return getResources(kind).reduce((total: number, resource: ResourceMeta) => {
+        const problemCount = problems
+          .filter(p => p.level === level)
```

Comment:
Can we do just one single filtering in here so we don't iterate twice the same array?